lib/internal/backend/repository/repository2.go

```diff
@@ -512,7 +504,6 @@ func (rb *RepositoryBackend) getLayerV2(layerID string, dockerURL *types.ParsedD
 			if err != nil {
 				return nil, nil, err
 			}
-			defer res.Body.Close()
```

Comment:
there's a problem however, response body closing is an issue in all of this routine, I'll add a defer